apps/aecore/src/aec_peer_connection.erl

```diff
@@ -0,0 +1,823 @@
+                        #{ <<"result">> => error, <<"reason">> => Reason }
+                end;
+            _ ->
+                #{ <<"result">> => error, <<"reason">> => <<"bad request">> }
+        end,
+    send_msg(S, ?MSG_GET_BLOCK_RSP, Msg),
+    S.
+
+handle_get_block_rsp(S, none, MsgObj) ->
+    lager:info("Peer ~p got unexpected GET_BLOCK_RSP - ~p", [peer_id(S), MsgObj]),
+    S;
+handle_get_block_rsp(S, {get_block, From}, MsgObj) ->
+    case maps:get(<<"result">>, MsgObj, <<"error">>) of
+        <<"ok">> ->
+            case maps:get(<<"block">>, MsgObj, undefined) of
+                Block0 when is_map(Block0) ->
+                    {ok, Block} = aec_blocks:deserialize_from_map(Block0),
+                    gen_server:reply(From, {ok, Block});
+                _ ->
+                    gen_server:reply(From, {error, no_block_in_response})
+            end;
+        <<"error">> ->
+            gen_server:reply(From, {error, maps:get(<<"reason">>, MsgObj)})
+    end,
+    remove_request_fld(S, get_block).
+
+%% -- Get Mempool --------------------------------------------------------------
+
+send_get_mempool(S = #{ status := error }, _From) ->
+    {reply, {error, disconnected}, S};
+send_get_mempool(S = #{ status := {connected, _ESock} }, From) ->
+    send_msg(S, ?MSG_GET_MEMPOOL, []),
+    {noreply, set_request(S, get_mempool, From)}.
+
+handle_get_mempool(S, _MsgObj) ->
+    {ok, Txs0} = aec_tx_pool:peek(infinity),
+    Txs = [ aetx_sign:serialize_to_binary(T) || T <- Txs0 ],
+    Msg = #{result => ok, txs => Txs},
+    send_msg(S, ?MSG_GET_MEMPOOL_RSP, Msg),
+    S.
+
+handle_get_mempool_rsp(S, none, MsgObj) ->
+    lager:info("Peer ~p got unexpected GET_MEMPOOL_RSP - ~p", [peer_id(S), MsgObj]),
+    S;
+handle_get_mempool_rsp(S, {get_mempool, From}, MsgObj) ->
+    case maps:get(<<"result">>, MsgObj, <<"error">>) of
+        <<"ok">> ->
+            case maps:get(<<"txs">>, MsgObj, undefined) of
+                Txs0 when is_list(Txs0) ->
+                    Txs = [ aetx_sign:deserialize_from_binary(T) || T <- Txs0 ],
+                    gen_server:reply(From, {ok, Txs});
+                _ ->
+                    gen_server:reply(From, {error, no_transactions_in_response})
+            end;
+        <<"error">> ->
+            gen_server:reply(From, {error, maps:get(<<"reason">>, MsgObj)})
+    end,
+    remove_request_fld(S, get_mempool).
+
+%% -- Send Block --------------------------------------------------------------
+
+send_send_block(S = #{ status := error }, _From, _Block) ->
+    {reply, {error, disconnected}, S};
+send_send_block(S = #{ status := {connected, _ESock} }, From, Block) ->
+    BlockMap = aec_blocks:serialize_to_map(Block),
+    send_msg(S, ?MSG_SEND_BLOCK, BlockMap),
+    {noreply, set_request(S, send_block, From)}.
+
+handle_send_block(S, MsgObj) ->
+    Msg =
+        case aec_blocks:deserialize_from_map(MsgObj) of
+            {ok, Block} ->
+                Header = aec_blocks:to_header(Block),
```

Comment:
Left over debug code?

Reply:
No we were logging that in the original design... I just kept it without thinking too much...